client/packages/common/src/hooks/useUrlQuery/useUrlQuery.ts

```diff
@@ -1,17 +1,47 @@
 import { useSearchParams } from 'react-router-dom';
 
 export interface UrlQueryObject {
-  [key: string]: string | number | boolean | RangeObject;
+  [key: string]: UrlQueryValue;
 }
 
-export interface RangeObject {
-  from?: string | number;
-  to?: string | number;
+export interface RangeObject<T> {
+  from?: T;
+  to?: T;
 }
 
+export type UrlQueryValue =
+  | string
+  | number
+  | boolean
+  | RangeObject<string | number>
+  | undefined;
+
 // CONSTANTS
 export const RANGE_SPLIT_CHAR = '_';
 
+// The following Regex are used to split range strings, either "Date/Time" or
+// "Number", and use RANGE_SPLIT_CHAR to separate. Both the start and end values
+// of the range are optional, but must be the same (date/dateTime/number) if
+// both present
+
+// Date/Date-Time range Regex
+// Will match ISO Date or Date/Time ranges
+// e.g. the following will all match
+// 2023-10-02 03:10_2023-10-03 02:10
+// 2023-10-02_2023-10-03
+// 2023-10-02 03:10_
+// _2023-10-03
+const dateRangeRegex = new RegExp(
+  `^(\\d{4}-\\d{2}-\\d{2})?_(\\d{4}-\\d{2}-\\d{2})?|(\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2})?${RANGE_SPLIT_CHAR}(\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2})?$`
+);
+
+// Number range Regex
+// A "number" can contain a negative (-) prefix, and a single decimal point
+// within it (which must be followed by additional digits)
+const numberRangeRegex = new RegExp(
+  `^(-?\\d+(\\.\\d+)?)?${RANGE_SPLIT_CHAR}(-?\\d+(\\.\\d+)?)?$`
```

Comment:
what if I'm (annoyingly) French and the decimal separator is `,`? 
actually, the inputs don't seem to use regional settings.. might be something else to add later 🤔

Reply:
Hmmm, good point. I guess we could pull the `.` out as a constant the way I have with `RANGE_SPLIT_CHAR`. But we don't really have any way to change that currently anyway (I guess it would need to be deduced from the locale settings), so let's leave it till we internationalise the inputs properly :)